app/src/main/java/org/p2p/wallet/striga/ui/TopUpWalletPresenter.kt

```diff
@@ -88,7 +83,7 @@ class TopUpWalletPresenter(
         }
     }
 
-    private suspend fun ensureNeededStrigaDataLoaded() {
+    private suspend fun ensureNeededStrigaDataLoaded(destination: StrigaUserStatusDestination) {
```

Comment:
im not sure that we can get correct destination if data is not loaded

Reply:
It is, you're right. I missed that data was loading before getting a destination.